contracts/BSCValidatorSet.sol

```diff
@@ -312,25 +346,49 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     return CODE_OK;
   }
 
-  function getValidators()external view returns(address[] memory) {
+  function getValidators() public view returns (address[] memory) {
     uint n = currentValidatorSet.length;
     uint valid = 0;
     for (uint i = 0;i<n;i++) {
-      if (!currentValidatorSet[i].jailed) {
+      if (isWorkingValidator(i)) {
         valid ++;
       }
     }
     address[] memory consensusAddrs = new address[](valid);
     valid = 0;
     for (uint i = 0;i<n;i++) {
-      if (!currentValidatorSet[i].jailed) {
+      if (isWorkingValidator(i)) {
         consensusAddrs[valid] = currentValidatorSet[i].consensusAddress;
         valid ++;
       }
     }
     return consensusAddrs;
   }
-  
+
+  function isWorkingValidator(uint index) public view returns (bool) {
+    if (index >= currentValidatorSet.length) {
+      return false;
+    }
+
+    // validatorExtraSet[index] should not be used before it has been init.
+    if (index >= validatorExtraSet.length) {
+      return !currentValidatorSet[index].jailed;
+    }
+
+    return !currentValidatorSet[index].jailed && !validatorExtraSet[index].isMaintaining;
+  }
+
+  function getMaintainingValidators() public view returns (address[] memory maintainingValidators) {
```

Comment:
Please hide this useless validator in mainnet.

Reply:
fixed